Parent/Parent/Dashboard/DashboardViewController.swift

```diff
@@ -154,77 +149,43 @@ class DashboardViewController: UIViewController, CustomNavbarProtocol {
 
     }
 
-    func setup() throws {
-        guard studentCollection == nil else { return }
-        viewState.isSiteAdmin = session.isSiteAdmin
-        studentCollection = try Student.observedStudentsCollection(session)
-        studentCountObserver = try Student.countOfObservedStudentsObserver(session) { [weak self] count in
-
-            // Check to see if all students were all removed during
-            // the current user session
-            var noMoreLinkedStudents = false
-            if count == 0,
-                let state = self?.viewState,
-                state.studentCount > 0,
-                state.isValidObserver {
-                noMoreLinkedStudents = true
+    func update() {
+        let pending = students.pending || presenter?.permissions.pending == true
+        if !pending {
+            if students.error != nil {
+                viewState.isValidObserver = false
+                return
             }
 
-            self?.viewState.studentCount = count
-            self?.configureStudentMenu()
+            viewState.studentCount = students.count
+            let showNotAParentModal = (!viewState.isValidObserver && !viewState.isSiteAdmin && presenter?.permissions.first?.becomeUser != true) || students.isEmpty
```

Comment:
I'm getting the "Not a Parent" modal when I log in as an account admin that can act as user but doesn't have any students. I think `|| students.isEmpty` is going to throw this case off because this use can `becomeUser` but `students.isEmpty` is true.